MaslowDue/system.cpp

```diff
@@ -395,113 +401,219 @@ uint8_t system_check_travel_limits(float *target)
 
 #ifdef MASLOWCNC
 
-// recalculate machine base dimensions from settings (in mm)
-  void recomputeGeometry(void)
+  void  chainToPosition(float aChainLength, float bChainLength, float *x,float *y ) {
+    _recomputeGeometry();
+
+    #if defined (KINEMATICS_DBG) && KINEMATICS_DBG > 0
+      Serial.print(F("Message: chainToPosition(), chainLength: "));
+      Serial.print(aChainLength);
+      Serial.print(',');
+      Serial.print(bChainLength);
+      Serial.print(F("; (guess) position: "));
+      Serial.print(*x);
+      Serial.print(',');
+      Serial.println(*y);
+    #endif
+
+    if (settings.kinematicsMode == KINEMATICS_MODE_TRIANGULAR) {
+      return triangular(aChainLength, bChainLength, x, y);
+    } else {
+      return forwardKinematics(aChainLength, bChainLength, x, y);
+    }
+  }
+
```

Comment:
I think this is wrong too or at least confusing.  "Forward" and "Triangular" are not modes and are not alternatives to each other.  I see you have at least in part renamed some of the functions inverse and forward functions.  But forward == chaintoposition and inverse == positiontochain.

I don't think you need to enable a setting for Kinematics mode (at least not yet) since we don't support quadrilateral.  You can leave it as a setting if you want, or just leave the place holder for potential future expansion.

Reply:
I understand. Per my last comment, my proposed solution is to introduce "forward" as a boolean setting. It would toggle between `triangular` and `triangularForward`. Future additions could, conceivably, introduce other modes if desired.

Reply:
Oh, I see.  Where triangular is just the simple geometric calculation assuming a straight chain.  Got it.  My apologies.

Maybe call it triangularForwardSimple or forwardSimple, as a more clear distinction.

Reply:
Cool, I was thinking of adding `Simple` to the non-forward as well :)